extensions/eclipse-che-theia-git-provisioner/src/node/git-configuration-controller.ts

```diff
@@ -0,0 +1,128 @@
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+/// <reference types='@theia/core/src/typings/nsfw/index'/>
+
+import { injectable, inject } from 'inversify';
+
+import { homedir } from 'os';
+import { resolve } from 'path';
+import { CheTheiaUserPreferencesSynchronizer } from '@eclipse-che/theia-user-preferences-synchronizer/lib/node/che-theia-preferences-synchronizer';
+import { writeFile, pathExists, createFile, readFile } from 'fs-extra';
+import * as ini from 'ini';
+import * as nsfw from 'nsfw';
+import { Disposable } from '@theia/core';
+
+export const GIT_CONFIG_PATH = resolve(homedir(), '.gitconfig');
+
+export interface UserConfiguration {
+    name: string | undefined;
+    email: string | undefined;
+}
+
+@injectable()
+export class GitConfigurationController {
+
+    @inject(CheTheiaUserPreferencesSynchronizer)
+    protected preferencesService: CheTheiaUserPreferencesSynchronizer;
+
+    protected preferencesHandler: Disposable | undefined;
+
+    protected gitConfigWatcher: nsfw.NSFW | undefined;
+
+    public async watchGitConfigChanges(): Promise<void> {
+        if (this.gitConfigWatcher) {
+            return;
+        }
+
+        const gitConfigExists = await pathExists(GIT_CONFIG_PATH);
+        if (!gitConfigExists) {
+            await createFile(GIT_CONFIG_PATH);
+        }
+
+        this.gitConfigWatcher = await nsfw(GIT_CONFIG_PATH, async (events: nsfw.ChangeEvent[]) => {
+            for (const event of events) {
+                if (event.action === nsfw.actions.MODIFIED) {
+                    const userConfig = await this.getUserConfigurationFromGitConfig();
+                    const preferences = await this.preferencesService.getPreferences();
+
+                    preferences['git.user.name'] = userConfig.name;
+                    preferences['git.user.email'] = userConfig.email;
+
+                    await this.preferencesService.setPreferences(preferences);
+                }
+            }
+        });
+        await this.gitConfigWatcher.start();
+    }
+
+    protected async getUserConfigurationFromGitConfig(): Promise<UserConfiguration> {
+        const gitConfigExists = await pathExists(GIT_CONFIG_PATH);
+        if (!gitConfigExists) {
+            return {} as UserConfiguration;
+        }
+
+        const gitConfigContent = await readFile(GIT_CONFIG_PATH, 'utf-8');
+        const gitConfig = ini.parse(gitConfigContent);
+
+        if (gitConfig.user === undefined) {
+            return {} as UserConfiguration;
+        }
+
+        const userName = gitConfig.user.name;
```

Comment:
Up to you, but it can be a bit simplier:
```typescript
const {name, email}  = gitConfig.user;

return { name, email }
```

Reply:
Thanks! Updated.